File: 898-transpose-matrix/transpose-matrix.java
class Solution {
    // Basic Matrix Traversal - TC: O(n*m), O(n*m) 
    public int[][] transpose(int[][] matrix) {
        int ROWS = matrix.length, COLS = matrix[0].length;
        int[][] res = new int[COLS][ROWS];

        for (int i = 0; i < ROWS; i++) {
            for (int j = 0; j < COLS; j++) {
                res[j][i] = matrix[i][j];
            }
        }
        return res;
    }
}

File: 898-transpose-matrix/transpose-matrix.py
class Solution:
    # Brute Force Solution - TC: O(m*n), SC: O(m*n)
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        ROWS, COLS = len(matrix), len(matrix[0])
        result = [[0 for x in range(ROWS)] for x in range(COLS)]

        for i in range(ROWS):
            for j in range(COLS):
                result[j][i] = matrix[i][j]
                
        return result